src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figuresNumber = 5;
+        int countFigure = random.nextInt(figuresNumber);
+        switch (countFigure) {
+            case 0:
+                return new Square(colorSupplier.getRandomColor(), getRandomInt());
+            case 1:
+                return new Rectangle(colorSupplier.getRandomColor(), getRandomInt(),
+                        getRandomDouble());
+            case 2:
+                return new RightTriangle(colorSupplier.getRandomColor(), getRandomDouble(),
+                        getRandomDouble());
+            case 3:
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        getRandomInt(),getRandomDouble(),getRandomDouble());
+            case 4:
+                return new Circle(colorSupplier.getRandomColor(), getRandomInt());
+            default:
+                return getDefaultFigure();
```

Comment:
Lets create 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction, this will increase code readability. (Remember private methods should follow after public as helper ones)